securedrop/journalist_app/sessions.py

```diff
@@ -0,0 +1,182 @@
+from datetime import datetime, timedelta
+from secrets import token_urlsafe
+from flask import Flask, Request, Response
+from typing import Optional, Any
+from flask.sessions import SessionInterface as FlaskSessionInterface
+from flask.sessions import SessionMixin, session_json_serializer
+from json.decoder import JSONDecodeError
+from redis import Redis
+from werkzeug.datastructures import CallbackDict
+from itsdangerous import URLSafeTimedSerializer, BadSignature
+
+
+class ServerSideSession(CallbackDict, SessionMixin):
+    """Baseclass for server-side based sessions."""
+
+    def __init__(self, sid: 'str', initial: 'Any' = None) -> 'None':
+        def on_update(self: ServerSideSession) -> None:
+            self.modified = True
+        CallbackDict.__init__(self, initial, on_update)
+        self.sid = sid
+        self.token = None
+        self.is_api = False
+        self.to_destroy = False
+        self.to_regenerate = False
+        self.modified = False
+
+
+class RedisSession(ServerSideSession):
```

Comment:
Does this need to be a separate class given that we will always be using redis? Can it be merged with ServerSideSession (or maybe the other way around)?